ml-agents/mlagents/trainers/behavior_id_utils.py

```diff
@@ -17,20 +17,14 @@ def from_name_behavior_id(name_behavior_id: str) -> "BehaviorIdentifiers":
         :returns: A BehaviorIdentifiers object.
         """
 
-        ids: Dict[str, int] = {}
+        team_id: int = 0
         if "?" in name_behavior_id:
-            name, identifiers = name_behavior_id.rsplit("?", 1)
-            if "&" in identifiers:
-                list_of_identifiers = identifiers.split("&")
-            else:
-                list_of_identifiers = [identifiers]
-
-            for identifier in list_of_identifiers:
-                key, value = identifier.split("=")
-                ids[key] = int(value)
+            name, team_and_id = name_behavior_id.rsplit("?", 1)
+            _, team_id_str = team_and_id.split("=")
```

Comment:
This is going to need to change again when we start passing the role (since split will return more that 2 values). I think we should just start using https://docs.python.org/3.7/library/urllib.parse.html for this, to avoid edge cases that we haven't considered. 

```
>>> from urllib.parse import urlparse, parse_qs
>>> parsed = urlparse("soccer?teamId=1&role=striker")
>>> parsed.path
'soccer'
>>> parse_qs(parsed.query)
{'teamId': ['1'], 'role': ['striker']}
```
Then we can extract team and role from the parse_qs result as needed.

Reply:
Yeah, I knew this would have to change when role was added.  Unfortunately, I'm not sure that's going to happen because Ervin and Arthur don't approve.

Good suggestion.

Reply:
Pushed.